autobahn/src/main/java/io/crossbar/autobahn/wamp/types/IEndpoint.java

```diff
@@ -0,0 +1,10 @@
+package io.crossbar.autobahn.wamp.types;
+
+import java.util.List;
+import java.util.Map;
+
+@FunctionalInterface
+public interface IEndpoint<R> {
+    // Tentative name, to be changed.
+    R get(List<Object> args, Map<String, Object> kwargs);
```

Comment:
the `get` here is basically the callback i.e. calling `get` will actually call the callback that we passed in.

We definitely need to change its from `get` to something sane.